[OSS] Add kwargs to fsspec reader/writer

ankitageorge opened the pull request:
Summary: Add kwargs to fsspec reader/writer. This will be used when reading/writing from huggingface because it needs a token to access the repositories

Test Plan: https://fburl.com/anp/agkrlas1 ability to read write to hf with fsspec

Differential Revision: D68738777




cc @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o @LucasLLC @MeetVadakkanchery @mhorowitz @pradeepfn @ekr0

mhorowitz commented on torch/distributed/checkpoint/_fsspec_filesystem.py:
```diff
@@ -59,7 +60,7 @@ def concat_path(
 
     def init_path(self, path: Union[str, os.PathLike]) -> Union[str, os.PathLike]:
-        self.fs, _ = url_to_fs(path)
+        self.fs, _ = url_to_fs(path, token=self.token)
```
`url_to_fs` takes `**kwargs` for generality.  Why do we only pass through token here?  This makes the code HF-specific.  If someone wants to use another fsspec provider which takes different arguments, we'd need to modify this code again.  In addition, the args are only needed here, so there's no reason to save them forever in a member.

Instead, I would suggest making init_path take the same arguments as url_to_fs, and pass them directly through.  Then modify the reader and writer ctors accordingly.

ankitageorge replied:
makes sense, updated

mhorowitz replied:
Looks much cleaner, thanks!

mhorowitz reviewed (approved):
Please make sure labels are set and lint is clean before landing.

ankitageorge committed:
[OSS] Add kwargs to fsspec reader/writer (#145845)

Summary:

Add kwargs to fsspec reader/writer. This will be used when reading/writing from huggingface because it needs a token to access the repositories

Test Plan: https://fburl.com/anp/agkrlas1 ability to read write to hf with fsspec

Reviewed By: mhorowitz

Differential Revision: D68738777
